src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject.ps1

```diff
@@ -0,0 +1,62 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Create an object to hold the cluster pool upgrade parameters.
+.Description
+Create an object to hold the cluster pool upgrade parameters.
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.ClusterPoolUpgrade
+.Link
+https://learn.microsoft.com/powershell/module/az.hdinsightonaks/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject
+#>
+function New-AzHdInsightOnAksClusterPoolAKSUpgradeObject {
+    [OutputType([Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.IClusterPoolUpgrade])]
+    [CmdletBinding(DefaultParameterSetName = 'Create', PositionalBinding = $false, SupportsShouldProcess)]
```

Comment:
This looks an in-memory cmdlet. `SupportsShouldProcess` is not necessary as no server side changes will be made.

Reply:
Removed unnecessary `SupportsShouldProcess `.